RELEASE_NOTES.md

```diff
@@ -1,3 +1,9 @@
+# Development Release 0.1.0 2018-05-17
```

Comment:
Awhile ago Dan changed the version number to 2.3.1 in ooi-config/anaconda-build/ion-functions but apparently didn't change it here. Seems we should make the version numbers match between them. I recommend changing it here to 2.4.0